internal/usecase/repository/domain_repository.go

```diff
@@ -355,27 +395,33 @@ func (r *domainRepository) createIpaDomain(
 	data *model.Ipa,
 ) (err error) {
 	if data == nil {
-		return internal_errors.NilArgError("data' of type '*model.Ipa")
+		err = internal_errors.NilArgError("data")
+		slog.Error(err.Error())
```

Comment:
Maybe I have taken the logs to the extreme, so any error is logged. Below some thoughts:

* (**paranoid** mode - current) any check that return an error, log the error message; when location is enabled that will provide kind of a trace stack on the log as every log message would be written in cascade. Downside is the calls to slog are spread on the code.
* (**more pragmatic**) log only for public function and methods (all the ones that start with upercase); this reduce the overload of calling to log the error message. The downside is, when the error happens in a private method or function, we don't have the exact location where it happens, but near enough; it could be a problem if many nested levels of private function and methods.
* (**centralized**) when the error is created, add some location information. This allow to know the location where the initial error happens, no log is written, and when the log middleware is going to log the information, print out the location inside the error. This simplify the logging centralizing everything on the middleware. Downside, if some panic happens before the log middleware write the log message with the error information and location, no trace will be written; another one is how to manage wrapped errors or what guidelines to follow to avoid that scenario. Upsides, the code is cleaner but we would need to define a common error type that will include the location information or even the stack trace when the error is created.
* (**other**) Any other option that I could be missing.

Which option should we use?

Reply:
I don't know which option. But my thoughts are:
- it makes sense to log when the issue is first encountered, e.g. created by us or by getting it from third party module
- I'm not sure about logging on every level, it could be done, but to be useful it would be good to have some kind of location logged with it so that I don't see 3 same messages in the log. I wonder if slog can add the location automatically.
- sometimes error is just an expected condition to handle - these might be avoid the not have too many "red herrings" in the log  (a general though - I'm not having any specific change from this PR on mind)

Maybe the pragmatic way might be to go with this PR as it is implemented and see how we like it. If not, it can be changed.